prefetch.bs

```diff
@@ -252,233 +325,283 @@ A <dfn export>cross-origin prefetch IP anonymization policy</dfn> has an <dfn ex
+        :  [=request/referrer policy=]
+        :: |entry|'s [=session history entry/document state=]'s [=document state/request referrer policy=]
+    1. If |documentResource| is a [=POST resource=], then:
+        1. Set |request|'s [=request/method=] to `` `POST` ``.
+        1. Set |request|'s [=request/body=] to |documentResource|'s [=POST resource/request body=].
+        1. [=header list/Set=] `` `Content-Type` `` to |documentResource|'s [=POST resource/request content-type=] in |request|'s [=request/header list=].
+    1. If |entry|'s [=session history entry/document state=]'s [=document state/reload pending=] is true, then set |request|'s [=request/reload-navigation flag=].
+    1. Otherwise, if |entry|'s [=session history entry/document state=]'s [=document state/ever populated=] is true, then set |request|'s [=request/history-navigation flag=].
+    1. If |hasTransientActivation| is true, then set |request|'s [=request/user-activation=] to true.
+    1. If |container| is non-null:
+        1. If |container| has a [=browsing context scope origin=], then set |request|'s [=request/origin=] to that [=browsing context scope origin=].
+        1. Set |request|'s [=request/destination=] and [=request/initiator type=] to |container|'s [=Element/local name=].
+    1. Return |request|.
+</div>
 
-    1. Let |shouldBlockNavigationRequest| be the following steps, given [=request=] |request|, [=string=] |navigationType| and [=environment=] |environment|:
-        1. [=Assert=]: |navigationType| is "`other`".
-        1. [=Assert=]: |request|'s [=request/reserved client=] is |environment|.
-        1. Let |purpose| be a <a spec="RFC8941">List</a> containing the <a spec="RFC8941">Token</a> "`prefetch`".
-        1. If |anonymizationPolicy| [=prefetch IP anonymization policy/requires anonymity=] for |request|, then:
-            1. Add a parameter whose key is <a for="Sec-Purpose prefetch" lt="anonymous-client-ip">"`anonymous-client-ip`"</a> and whose value is true to the "`prefetch`" token in |purpose|.
-            1. The user agent must use a [=connection=] which anonymizes the client IP address (e.g., using a proxy) when fetching |request|, or return "`Blocked`".
+<hr>
 
-                <p class="issue">At the moment, how IP anonymization is achieved is handwaved. This will probably be done in an [=implementation-defined=] manner using some kind of proxy or relay. Ideally this would be plumbed down to [=obtain a connection=], and possibly even the mechanism could be further standardized.</p>
-        1. [=header list/Set a structured field value=] given (<a http-header>`` `Sec-Purpose` ``</a>, |purpose|) in |request|'s [=request/header list=].
+<div algorithm="attempt to populate the history entry's document">
+    In <a spec=HTML>attempt to populate the history entry's document</a>, replace the step which invokes <a spec=HTML>create navigation params by fetching</a> with the following:
```

Comment:
To be clear, this means we're allowing prefetches to be used to fulfill traversals, not just push/replace navigations. I think that's OK but I wanted to make sure you agree. (If not, we should probably add some sort of guard.)

Reply:
Fun!

Looks like the test got reverted; I hope we can add it back. It's weird that it would be failing on Macs only?